src/gosling-track/gosling-track.ts

```diff
@@ -640,36 +636,33 @@ function GoslingTrack(HGC: import('@higlass/types').HGC, ...args: any[]): any {
         /**
          * Show visual cue during waiting for visualizations being rendered.
          */
-        drawLoadingCue() {
-            if (this.fetching.size) {
-                const margin = 6;
-
-                // Show textual message
-                const text = `Fetching... ${Array.from(this.fetching).join(' ')}`;
-                this.loadingText.text = text;
-                this.loadingText.x = this.position[0] + this.dimensions[0] - margin / 2.0;
-                this.loadingText.y = this.position[1] + this.dimensions[1] - margin / 2.0;
-
-                // Show background
-                const metric = HGC.libraries.PIXI.TextMetrics.measureText(text, this.loadingTextStyleObj);
-                const { width: w, height: h } = metric;
-
-                this.loadingTextBg.clear();
-                this.loadingTextBg.lineStyle(1, colorToHex('grey'), 1, 0.5);
-                this.loadingTextBg.beginFill(colorToHex('white'), 0.8);
-                this.loadingTextBg.drawRect(
-                    this.position[0] + this.dimensions[0] - w - margin - 1,
-                    this.position[1] + this.dimensions[1] - h - margin - 1,
-                    w + margin,
-                    h + margin
-                );
+        showLoadingCue(text: string) {
+            const margin = 6;
+
+            this.loadingText.text = text;
+            this.loadingText.x = this.position[0] + this.dimensions[0] - margin / 2.0;
+            this.loadingText.y = this.position[1] + this.dimensions[1] - margin / 2.0;
+
+            // Show background
+            const metric = HGC.libraries.PIXI.TextMetrics.measureText(text, this.loadingTextStyleObj);
+            const { width: w, height: h } = metric;
+
+            this.loadingTextBg.clear();
+            this.loadingTextBg.lineStyle(1, colorToHex('grey'), 1, 0.5);
+            this.loadingTextBg.beginFill(colorToHex('white'), 0.8);
+            this.loadingTextBg.drawRect(
+                this.position[0] + this.dimensions[0] - w - margin - 1,
+                this.position[1] + this.dimensions[1] - h - margin - 1,
+                w + margin,
+                h + margin
+            );
 
-                this.loadingText.visible = true;
-                this.loadingTextBg.visible = true;
-            } else {
+            this.loadingText.visible = true;
+            this.loadingTextBg.visible = true;
+            return () => {
                 this.loadingText.visible = false;
                 this.loadingTextBg.visible = false;
-            }
+            };
```

Comment:
I wonder if this could be even further refactored to add/remove the loading text from `this.pLabel`.